enhancements/insights/conditional-data-gathering.md

```diff
@@ -0,0 +1,199 @@
+and consist of only alphanumeric characters
+- Only implemented gathering functions can be used
+- Namespace from `gather_logs_of_namespace` function should be a string of length between 1 and 128, 
+match the next regular expression `^[a-zA-Z]([a-zA-Z0-9\-]+[\.]?)*[a-zA-Z0-9]$` and start with `openshift-`
+- Tail lines from `gather_logs_of_namespace` function should be an integer between 1 and 4096
+- Namespace from `gather_image_streams_of_namespace` function should be a string of length between 1 and 128, 
+match the next regular expression `^[a-zA-Z]([a-zA-Z0-9\-]+[\.]?)*[a-zA-Z0-9]$` and start with `openshift-`
+
+If anything fails to be validated, the config is rejected and an error is written to the logs.
+The PR with validation on operator side - https://github.com/openshift/insights-operator/pull/470
+
+5. Insights Operator goes rule by rule and collects the requested data if corresponding conditions are met
+
+### User Stories
+
+For OpenShift engineers it will be easier to get necessary information when some problem occurs (like alert is firing).
+Getting such information now means changing operator's code which is especially painful with older versions of clusters.
+When the data is not needed anymore, the rule can be simply removed from the repo.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Empty
+
+### Risks and Mitigations
+
+The potential risks could come from an attacker spoofing the config somehow (if they got access to the repo),
+all they could do is let the insights operator collect more data and send it to the c.r.c, but because of validation
+on the operator side, the potential of collecting data is limited. For example, 
+we check that namespaces start with `openshift-`, we're also limiting the amount of potentially collected data by,
+for example, introducing the limit for amount of collected logs. 
+
+Also in the regular workflow, changing the config involves going through the repo's CI (JSON schema validator) 
+and probably a simple review process. 
+
+Not really a risk, but in case the service is not available or provides invalid config for the operator,
+we would just have less data in the archives (everything except conditional gatherer's data), 
+an error would be written to the logs and to the metadata which would allow us to know 
+that something is broken and we need to bring the service back.
+
+## Design Details
+
+Empty
+
+### Open Questions [optional]
+
+Empty
+
+### Test Plan
+
+The conditional gatherer is covered with unit tests considering many different scenarios. 
+Later there will be integration tests as well.
+
+### Graduation Criteria
+
+Empty
+
+### Upgrade / Downgrade Strategy
+
+The conditional gatherer itself (e.g. adding new/removing old conditions or gathering functions) is 
+modified through a standard process like all other operator's code. Only the rules to collect X with Y params on Z
+are updated through the new process described above.
```

Comment:
Does this enhancement need to cover schema changes for the config files, or is that addressed in another document already?

Reply:
It's not, I will add it to the doc.